src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/SpeakerWorkerService.java

```diff
@@ -148,14 +151,25 @@ private void completeChunkedResponse(String key, List<ChunkedInfoMessage> messag
 
         if (pending != null && pending.getPayload() != null) {
             if (pending.getPayload() instanceof DumpRulesForSwitchManagerRequest) {
-                FlowDumpResponse entries = new FlowDumpResponse(
-                        messages.stream()
-                                .map(InfoMessage::getData)
-                                .map(SingleFlowDumpResponse.class::cast)
-                                .map(SingleFlowDumpResponse::getFlowSpeakerData)
-                                .collect(Collectors.toList()));
-                InfoMessage response = new InfoMessage(entries, key, pending.getCookie());
-                carrier.sendResponse(key, response);
+                List<FlowDumpResponse> responses = messages.stream()
```

Comment:
Seems like a good place for generic method or Strategy pattern.

Reply:
1. I don't remember if the one can use generic/parametrized methods without making the class parametrized. If no - then you're right, my first suggestion doesn't make sense.
2. Can't think of a good usage of Strategy here. It was a bad suggestion.